systemtest/src/test/java/io/strimzi/systemtest/SecurityST.java

```diff
@@ -422,6 +425,66 @@ private String createSecret(String resourceName, String secretName, String keyNa
         return certAsString;
     }
 
+    @Test
+    void testCertRenewalInMaintenanceWindow() throws Exception {
+        String secretName = CLUSTER_NAME + "-cluster-ca-cert";
+        long maintenanceWindowDuration = 4;
+        long currentMinute = LocalDateTime.now().getMinute();
+        long maintenanceStart = currentMinute + maintenanceWindowDuration;
+        if (maintenanceStart > 59) {
+            maintenanceStart = maintenanceStart - 60;
+        }
+        long maintenanceStop = maintenanceStart + maintenanceWindowDuration;
+
+        String maintenanceWindowCron = "* " + maintenanceStart + "-" + maintenanceStop + " * * * ? *";
+        LOGGER.info("Maintenance window is: {}", maintenanceWindowCron);
+        testMethodResources.kafkaEphemeral(CLUSTER_NAME, 3, 1)
+                .editSpec()
+                .editKafka()
+                .editListeners()
+                .withNewKafkaListenerExternalNodePort()
+                .withTls(false)
+                .endKafkaListenerExternalNodePort()
+                .endListeners()
+                .endKafka()
+                .addNewMaintenanceTimeWindow(maintenanceWindowCron)
+                .endSpec().done();
+
+        Map<String, String> kafkaPods = StUtils.ssSnapshot(kafkaStatefulSetName(CLUSTER_NAME));
+
+        LOGGER.info("Annotate secret {} with secret force renew annotation", secretName);
+        Secret secret = new SecretBuilder(kubeClient().getSecret(secretName))
+                .editMetadata()
+                .addToAnnotations("strimzi.io/force-renew", "true")
+                .endMetadata().build();
+        kubeClient().patchSecret(secretName, secret);
+
+        LOGGER.info("Wait until maintenance windows starts");
+        long finalMaintenanceStart = maintenanceStart;
+        TestUtils.waitFor("Wait until maintenance window start",
+            Constants.GLOBAL_POLL_INTERVAL, Duration.ofMinutes(maintenanceWindowDuration).toMillis(),
+            () -> LocalDateTime.now().getMinute() >= finalMaintenanceStart);
+
+        // Get current hour before rolling update for assert
+        long maintenanceStopHour = LocalDateTime.now().getHour();
+        assertThat("Rolling update was performed out of maintenance window!", kafkaPods, is(StUtils.ssSnapshot(kafkaStatefulSetName(CLUSTER_NAME))));
+
+        LOGGER.info("Wait until rolling update is triggered during maintenance window");
+        StUtils.waitTillSsHasRolled(kafkaStatefulSetName(CLUSTER_NAME), 3, kafkaPods);
+
+        assertThat("Rolling update wasn't performed in correct time", compareMaintenanceTime(maintenanceStart, maintenanceStopHour));
+        waitForClusterAvailability(NAMESPACE);
+    }
+
+    private boolean compareMaintenanceTime(long minute, long hour) {
```

Comment:
Tests, where data is used, are always very complex. And compare data as _long_, can confuse person why will update this test in the future. I guess the better way is having the variables as _LocalDateTime_ in this ST. And in that case, you are able to use _compareTo_ method for better time calculation. Java doc - https://docs.oracle.com/javase/8/docs/api/java/time/LocalDateTime.html#compareTo-java.time.chrono.ChronoLocalDateTime-